src/detail-page/components/charts/DevOpsChart.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { FC } from 'react';
+import { Pie } from 'react-chartjs-2';
+import { devOps } from '../../../data/charts';
+import { chartColors } from '../../../data/charts/chartColors';
+import { Chart } from './Chart';
+import { notEmpty } from '../../../shared/utils/helpers';
+
+const { data: sourceData, labelText, source, title: chartTitle, type } = devOps;
+
+const chartLabels = sourceData.map(({ label }) => label);
+const chartValues = sourceData.map(({ value }) => value);
+const chartBackgroundColors = sourceData
+  .map(({ color }) => (color ? chartColors[color].light : null))
+  .filter(notEmpty);
+const chartBorderColors = sourceData
+  .map(({ color }) => (color ? chartColors[color].dark : null))
+  .filter(notEmpty);
+
+const areColorsDefined = !sourceData.some(({ color }) => color === null);
+
+const ChartData: Chart.ChartData = {
+  labels: chartLabels,
+  datasets: [
+    {
+      data: chartValues,
+      borderWidth: 1,
+      ...(labelText && {
+        label: labelText,
+      }),
+      ...(areColorsDefined && {
+        backgroundColor: chartBackgroundColors,
+        borderColor: chartBorderColors,
+      }),
+    },
+  ],
+};
```

Comment:
It's definitely not the best way to do that but chart.js types required not null/undefined value for any datasets key so I have to filter/cast it.